### DISCUSSION TITLE
[Question]: Does this stack include Zotero Web?

### DISCUSSION THREAD

**Discussion by MadFisa** [uniuuu/zotprime#46] [Ideas]
Hi,

Basically what the title says. Does this stack include the Zotero Web i.e the web app that you get on logging into zotero.org ?
From what I could understand from my limited knowledge and going through the repository is that this repo primarily the data server that will allow you to connect the desktop client to a selfhosted server and doesn't contain Zotero Web.

Is it possible to have a self-hosted Zotero Web added to the stack?

Thank you for your great work btw. I remember searching for something similar to what zotprime provides a while back and didn't find anything useable.

**Comment by Amniotic3** [comment]
If you are primarily using zotero web to manage users, `bin/create-user.sh` under the production branch can be used to add users.

**Comment by uniuuu** [comment]
> Hi,
> 
> Basically what the title says. Does this stack include the Zotero Web i.e the web app that you get on logging into zotero.org ? From what I could understand from my limited knowledge and going through the repository is that this repo primarily the data server that will allow you to connect the desktop client to a selfhosted server and doesn't contain Zotero Web.
> 
> Is it possible to have a self-hosted Zotero Web added to the stack?
> 
> Thank you for your great work btw. I remember searching for something similar to what zotprime provides a while back and didn't find anything useable.

> If you are primarily using zotero web to manage users, `bin/create-user.sh` under the production branch can be used to add users.


Hi @MadFisa @Amniotic3 
It's not in stack. 
Could you please share a link to Zotero Web repository ?

**Reply by MadFisa** [reply]
@uniuuu 

Here is the link for weblibrary : https://github.com/zotero/web-library

I should clarify that when I said Zotero Web, what I meant is that the part of the Zotero website that lets you view and read your library straight from the browser. My use case is that I may have to access Zotero from computers that may not have Zotero client installed for basic tasks, such as just viewing the pdf with the annotations.

I set up the above-linked website in a docker container for some testing. Looks like it just has a web library part of the Zotero website and nothing else (just as the repo name suggests).
The problem is there is that there is no user login/authentication. One has to generate a userid and api key and give it manually the index.html file if I understand it correctly. Is there a way to do that using Zotprime (i.e. generate an api key)?
Even better would be to have a login page for the user and the rest be automatically taken care of in the background, but don't know how much work it would involve.

**Reply by MadFisa** [reply]
I think the rest of the website is here: https://github.com/zotero/web-components/tree/master

I  will try setting it up when I get some time. But it doesn't look like there is much documentation there.

**Reply by uniuuu** [reply]
@MadFisa 
The quickest solution with lower effort would be using https://github.com/linuxserver/docker-zotero which runs desktop client in web browser.

**Reply by uniuuu** [reply]
> I should clarify that when I said Zotero Web, what I meant is that the part of the Zotero website that lets you view and read your library straight from the browser. My use case is that I may have to access Zotero from computers that may not have Zotero client installed for basic tasks, such as just viewing the pdf with the annotations.

@MadFisa If you need only to view and read then next project can fit the requirement https://whiskyechobravo.github.io/kerko/latest/   - see the Demo via link shared there.  Repository: https://github.com/whiskyechobravo/kerkoapp

**Reply by MadFisa** [reply]
@uniuuu I was also considering running the client in the web browser, it would work for me now since I'm a single user.
But, in a multi-user situation, it may not be ideal.

I don't know about kerkoapp before. It looks interesting, thank you for sharing.

**Comment by uniuuu** [comment]
> @uniuuu
> 
> Here is the link for weblibrary : https://github.com/zotero/web-library
> 
> I should clarify that when I said Zotero Web, what I meant is that the part of the Zotero website that lets you view and read your library straight from the browser. My use case is that I may have to access Zotero from computers that may not have Zotero client installed for basic tasks, such as just viewing the pdf with the annotations.
> 
> I set up the above-linked website in a docker container for some testing. Looks like it just has a web library part of the Zotero website and nothing else (just as the repo name suggests). The problem is there is that there is no user login/authentication. One has to generate a userid and api key and give it manually the index.html file if I understand it correctly. Is there a way to do that using Zotprime (i.e. generate an api key)? Even better would be to have a login page for the user and the rest be automatically taken care of in the background, but don't know how much work it would involve.



> I think the rest of the website is here: https://github.com/zotero/web-components/tree/master
> 
> I will try setting it up when I get some time. But it doesn't look like there is much documentation there.

Hi @MadFisa 
Good day.
Was it possible to assemble web page from these components for you?

Website
├── web-library (Main SPA)
│   ├── Uses: React, Redux, Zotero API
│   └── Imports: web-common (as submodule)
├── web-components (UI Elements)
│   ├── Uses: React, standalone components
│   └── Imports: web-common (as submodule)
└── web-common (Shared Foundation)
    ├── components/
    ├── hooks/
    └── utils/ 

Do you think this above is correct structure, or more components needed ?

**Comment by uniuuu** [comment]
> Hi,
> 
> Basically what the title says. Does this stack include the Zotero Web i.e the web app that you get on logging into zotero.org ? From what I could understand from my limited knowledge and going through the repository is that this repo primarily the data server that will allow you to connect the desktop client to a selfhosted server and doesn't contain Zotero Web.
> 
> Is it possible to have a self-hosted Zotero Web added to the stack?
> 
> Thank you for your great work btw. I remember searching for something similar to what zotprime provides a while back and didn't find anything useable.

>  I'm even wondering if it's possible to replicate the functionality of zotero's official website, such as account registration, password changes, maintenance of group permissions, and so on. The command line approach is still pretty geeky, and other members of the team have been giving me feedback on whether or not it could be UI-ized.

Hi @MadFisa @Amniotic3 
The minimal web admin panel has been released. 
Clone latest code and try it out with `docker compose --profile admin -f docker-compose.yml -f docker-compose-localhost.yml -f docker-compose-dev.yml up -d --build` or `docker compose --profile admin -f docker-compose.yml -f docker-compose-dev.yml up -d --build` if in VM/LXD/VPC

**Comment by uniuuu** [comment]
Further on in please continue in respective release discussion  https://github.com/uniuuu/zotprime/discussions/54
